.github/workflows/nf-test.yml

```diff
@@ -0,0 +1,162 @@
+name: Run nf-test
+on:
+  pull_request:
+    branches: [master]
+  merge_group:
+    types: [checks_requested]
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      runners:
+        description: "Runners to test on"
+        type: choice
+        options:
+          - "ubuntu-latest"
+          - "self-hosted"
+        default: "self-hosted"
+
+# Cancel if a newer run is started
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  NXF_SINGULARITY_CACHEDIR: ${{ github.workspace }}/.singularity
+  NXF_SINGULARITY_LIBRARYDIR: ${{ github.workspace }}/.singularity
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  nf-test:
+    runs-on: ${{ github.event.inputs.runners || 'self-hosted' }}
+    # NOTE I think this is the cleanest way to get them organized
+    # process | conda | 1
+    # process | conda | 2
+    # process | conda | 3
+    # process | docker_self_hosted | 1
+    # ...
+    # workflow | singularity | 3
+    name: ${{ matrix.filter }} | ${{ matrix.profile }} | ${{ matrix.shard }}
+    strategy:
+      fail-fast: false
+      matrix:
+        filter: [process, workflow]
+        profile: [conda, docker_self_hosted, singularity]
+        shard: [1, 2, 3]
```

Comment:
> Is it possible to set it dynamically based on number of tests with an estimate of how many we can pack onto a machine? e.g. n_tests %% 5?

That would be the dream. But yeah I also wanted to avoid the dynamic number of jobs. 

3 was just a random number that I picked. Trying to walk the line between runners sitting there, and not enough jobs.

Hot take: What if we just did a `--dryRun` at the beginning of the shard and if there's no jobs, exit before we set anything else up.